src/katgpucbf/fgpu/engine.py

```diff
@@ -377,11 +377,30 @@ def capacity(self) -> int:  # noqa: D401
         # basically, the number of spectra.
         return self.spectra.shape[0] * self.spectra.shape[2]
 
+    @property
+    def next_timestamp(self) -> int:  # noqa: D401
+        """Timestamp of the next chunk afer this one."""
+        return self.timestamp + self.capacity * self.spectra_samples
+
     @property
     def pols(self) -> int:  # noqa: D401
         """Number of polarisations."""
         return self.spectra.shape[3]
 
+    @property
```

Comment:
I feel like this should be a method rather than a property, because it takes O(n) time rather than O(1).

Reply:
If I remove this, flake8 gives me:
```
src/katgpucbf/fgpu/engine.py:921: error: Function "all_present" could always be true in boolean context  [truthy-function]
```
but I don't see how that's the case, I haven't used a literal.

Though I seem to recall mypy thought the left side of the AND was a literal too 🤔

Reply:
Duh. My bad.